packages/nuxt/src/components/plugins/tree-shake.ts

```diff
@@ -40,6 +51,21 @@ export const TreeShakeTemplatePlugin = (options: TreeShakeTemplatePluginOptions)
         regexpMap.set(components, [new RegExp(`(${clientOnlyComponents.join('|')})`), new RegExp(`^(${clientOnlyComponents.map(c => `(?:(?:_unref\\()?(?:_component_)?(?:Lazy|lazy_)?${c}\\)?)`).join('|')})$`), clientOnlyComponents])
       }
 
+      const page = !import.meta.dev && options.routeData.pages.find(p => p.file === id)
+      // A page component is being tree shaken
+      if (page) {
+        const matchedRules = ruleMatcher!.matchAll(page.path).reverse()
+        const resolvedRoute = defu({} as Record<string, any>, ...matchedRules) as NitroRouteConfig
+        if (resolvedRoute) {
+          // page is SPA
+          if (resolvedRoute.ssr === false) {
+            return {
+              code: '',
```

Comment:
- do we need to keep source maps for debugging?
- emitting an empty Vue SFC seems a bit strange, it feels like this should be happening at the router level, but non-blocking if its working